radix-transactions/src/manifest/static_resource_movements/effect.rs

```diff
@@ -31,338 +38,602 @@ pub enum InvocationReceiver {
-                                    ValidatorMethod::Stake(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::Unstake(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::ClaimXrd(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::UpdateKey(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::UpdateFee(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::UpdateAcceptDelegatedStake(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::AcceptsDelegatedStake(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::TotalStakeXrdAmount(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::TotalStakeUnitSupply(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::GetRedemptionValue(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::SignalProtocolUpdateReadiness(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::GetProtocolUpdateReadiness(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::LockOwnerStakeUnits(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::StartUnlockOwnerStakeUnits(input) => {
-                                        input.output(details)
-                                    }
-                                    ValidatorMethod::FinishUnlockOwnerStakeUnits(input) => {
-                                        input.output(details)
-                                    }
-                                }
-                            }
+                    AccessControllerBlueprintInvocations::Method(invocation) => match invocation {
```

Comment:
Or if we want to split this up, something like:

```rust
create_match_on_access_controller_invocations! {
    invocation,
    (input) => input.output(details),
}
```

Which could output:
```rust
match invocation {
    AccessControllerBlueprintInvocations::Function(invocation) => match invocation {
        // ...
        _ => unreachable!("Avoid compile errors on empty enums"),
    }
    AccessControllerBlueprintInvocations::Method(invocation) => match invocation {
        // ...
        _ => unreachable!("Avoid compile errors on empty enums"),
    }
}
```

Reply:
I'm totally open to the idea of having the macro that generates the typed invocation enum also implement `StaticInvocationResourcesOutput` on it. It just felt like it mixes and tangles these two concepts together which we may want to keep separate. As an example, if we move the typed invocations to `radix-engine-interface` and keep the `StaticInvocationResourcesOutput` trait in the `radix-transactions` crate we would need to add a `radix-transactions` dependency to the `radix-engine-interface` crate to get the macro to implement the trait, or we would need to move the trait as well to `radix-engine-interface` since `radix-transactions` already has a `radix-engine-interface` dependency. Perhaps this example highlights what I mean when I say that it tangles these two concepts together where they now sort of need to move in tandem with one another. So, even though it feels very useful to have the macro handle this implementation, i'm somewhat uncertain about it.

Reply:
As mentioned, I can see the benifit of doing it in the macro and I'm open to the idea if you'd like to discuss it.

Reply:
I agree on the separation between the typed invocation and this trait.

I'm not suggesting we output a macro which implements the trait.

I’m suggesting we output a macro which generates this massive nested match statement, which can then be used to more easily implement this trait - or indeed other traits. (e.g. to output side effects or further constraints).

Reply:
We can discuss on a call tomorrow